bin/nrnivmodl_makefile_cmake.in

```diff
@@ -38,8 +38,11 @@ _cm =,
 INCLUDES = -I. $(INCFLAGS) $(UserINCFLAGS) -I$(incdir)
 INCLUDES += $(if @MPI_C_INCLUDE_PATH@, -I$(subst ;, -I,@MPI_C_INCLUDE_PATH@),)
 
-CC ?= @CMAKE_C_COMPILER@
-CXX ?= @CMAKE_CXX_COMPILER@
+# CC and CXX are implicitly defined. We check against cc, which is never (hopefully) user defined
+ifeq ($(CC), cc)
+    CC = @CMAKE_C_COMPILER@
+    CXX = @CMAKE_CXX_COMPILER@
+endif
```

Comment:
@ferdonline : thinking bit more : what if we do following: 

* If we are building via setup.py, additionally we do following: Instead of using full path of the compiler i.e. `@CMAKE_C_COMPILER@`, we use just compiler name i.e. gcc/g++ for linux or cc/cxx for osx. This way, even user will set the CC, it won't be fatal error but will use compiler that will be exist anyway.

This is just to be extra careful. What do you think?

(For implementation : we can set extra CMake option from setup.py `-DNRN_COMPILER_FULL_PATH=OFF` and then CMake internally can set new two variables for C/CXX which can be used above).

Reply:
It was good to think/learn a bit more about gnu make. The `$(origin` function exists for these cases :)

Reply:
I must say I run out of patience while reading makefile details! 😀 

This is working perfectly! thank you @ferdonline !